ports/rp2/modmachine.c

```diff
@@ -80,10 +82,19 @@ STATIC mp_obj_t machine_bootloader(void) {
 }
 MP_DEFINE_CONST_FUN_OBJ_0(machine_bootloader_obj, machine_bootloader);
 
-STATIC mp_obj_t machine_freq(void) {
-    return MP_OBJ_NEW_SMALL_INT(clock_get_hz(clk_sys));
+STATIC mp_obj_t machine_freq(size_t n_args, const mp_obj_t *args) {
+    if (n_args == 0) {
+        return MP_OBJ_NEW_SMALL_INT(clock_get_hz(clk_sys));
+    } else {
+        mp_int_t freq = mp_obj_get_int(args[0]);
+        if (set_sys_clock_khz(freq / 1000, false)) {
```

Comment:
Bit more concise and clear (at least to me):
```
if (!set_sys_clock_khz(freq / 1000, false)) {
    mp_raise_ValueError(MP_ERROR_TEXT("valid range 12-270MHz"));
}
return mp_const_none;
``` 
?

Reply:
I prefer to  show both wings or a if/else statement.

Reply:
But I changed it. Don't care about it.